app/actions/websocket/teams.ts

```diff
@@ -49,3 +53,43 @@ export async function handleLeaveTeamEvent(serverUrl: string, msg: any) {
         }
     }
 }
+
+export async function handleUpdateTeamEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const team = JSON.parse(msg.data.team) as Team;
+        database.operator.handleTeam({
+            teams: [team],
+            prepareRecordsOnly: false,
+        });
+    } catch {
+        // Do nothing
+    }
+}
+
+export async function handleTeamAddedEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const teamId = msg.data.team_id;
+    const {teams, memberships: teamMemberships} = await fetchMyTeam(serverUrl, teamId, false);
```

Comment:
Do we want to batch this database operations with the ones from fetchRolesIfNeeded?

Reply:
done